apps/ejabberd/src/mongoose_cassandra.erl

```diff
@@ -0,0 +1,121 @@
+%%%-------------------------------------------------------------------
+%%% @author Uvarov Michael <arcusfelis@gmail.com>
+%%% @copyright 2016 Erlang Solutions, Ltd.
+%%% @doc Cassandra functions
+%%% @end
+%%%-------------------------------------------------------------------
+-module(mongoose_cassandra).
+-export([start/0, stop/0]).
+-export([now_timestamp/0]).
+-export([status/0, get_pools/0]).
+
+-include_lib("ejabberd/include/ejabberd.hrl").
+
+-callback prepared_queries() -> list({term(), string()}).
+
+start() ->
+    [start_pool(Pool) || Pool <- get_pools()].
+
+start_pool({PoolName, PoolConfig}) ->
+    PoolConfig1 = deduplicate_proplist(expand_config(PoolConfig)),
+    mongoose_cassandra_sup:start(PoolName, PoolConfig1).
+
+expand_config(PoolConfig) ->
+    expand_servers(expand_credentials(PoolConfig))
+    ++ [{servers, [{"localhost", 9042, 1}]},
+        {socket_options, [{connect_timeout, 4000}]},
+        {keyspace, "mongooseim"},
+        {credentials, undefined}].
+
+expand_servers(PoolConfig) ->
+    Address     = proplists:get_value(address, PoolConfig),
+    Servers     = proplists:get_value(servers, PoolConfig, []),
+    expand_servers(Address, Servers, PoolConfig).
+
+expand_servers(undefined, undefined, PoolConfig) ->
+    [{servers, [{"localhost", 9042, 1}]}];
+expand_servers(undefined, Servers, PoolConfig) ->
+    PoolConfig;
+expand_servers(Address, undefined, PoolConfig) ->
+    DefaultPort = proplists:get_value(port, PoolConfig, 9042),
+    PoolSize    = proplists:get_value(pool_size, PoolConfig, 5),
+    Servers = expand_address(Address, DefaultPort, PoolSize),
+    [{servers, Servers} | PoolConfig];
+expand_servers(_Address, _Servers, PoolConfig) ->
+    ?ERROR_MSG("issue=config_error, details=\"Both servers and address options "
```

Comment:
servers option is used by driver. It's not self-descriptive and people will make mistakes trying to configure it.

```erlang
{servers, [{"localhost", 9042, 1}]}
```

But I want to use address to be similar to riak backend and make it easier to use

```erlang
{address, "localhost"}
```

So, in case someone still wants to use servers to configure number of workers per server - he can do this. But it's not documented feature.